vllm/engine/arg_utils.py

```diff
@@ -390,20 +391,21 @@ def add_cli_args(
             default=False,
             help='If True, the prefill requests can be chunked based on the '
             'max_num_batched_tokens')
-
         parser.add_argument(
             '--speculative-model',
             type=str,
             default=None,
             help=
             'The name of the draft model to be used in speculative decoding.')
-
         parser.add_argument(
             '--num-speculative-tokens',
             type=int,
             default=None,
             help='The number of speculative tokens to sample from '
             'the draft model in speculative decoding')
+        parser.add_argument("--embedding-mode",
```

Comment:
This idea was proposed before. The idea is that xxxForCausalLM is usually for generation. But we need to investigate if there will be edge cases that use xxxModel for fine tuning instead of embedding.

Reply:
HF has the following standards:
- `XXXModelForCausalLM` <==>  decoder with LMHead
- `XXXModelForConditionalLM` <==> encoder-decoder with LMHead
- `XXXModel` <==> embedding

So, we are good to go with this assumption. Plus we already have a registry of explicit models we support so I think this is okay

Reply:
Done

Reply:
Yup. It got mixed up during rebasing.